### PR TITLE
DHFPROD-2905: Support 9.0-10 and above, 10.0 nightly > 06/01

### PR DESCRIPTION
**Pull request by srinathgit**
Support for 4.x-develop on 9.0-10 and above, 10.0 nightly > 06/01

### CODE REVIEW

**Comment by josvanroosmalen** [comment]
Can we get this also fixed in 4.2.2 (e.g. 4.2.3 or 4.2.2-1 or whatever)

**Comment by josvanroosmalen** [comment]
I am very critical on this change. Why do you support 10.0 Nightlies, but not 10.0-2 once that hits the ground. This does not make any sense!?

**Comment by sbayatpur** [comment]
4.x should work on 9.0-10 and future versions of 9, but users should upgrade to 5 if they want to use MarkLogic 10

File: marklogic-data-hub/src/main/java/com/marklogic/hub/impl/DataHubImpl.java [hunk 1]
```diff
@@ -246,53 +246,42 @@ public boolean isServerVersionValid(String versionString) {
                 versionString = versions.getMarkLogicVersion();
             }
             int major = Integer.parseInt(versionString.replaceAll("([^.]+)\\..*", "$1"));
-            if (major < 9) {
+            if (major != 9) {
                 return false;
             }
             boolean isNightly = versionString.matches("[^-]+-(\\d{4})(\\d{2})(\\d{2})");
             //Support any 9.0 version > 9.0-5 and all 10.0 versions.
-            if (major == 9) {
```
Comment (bsrikan) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/impl/DataHubImpl.java]: We probably have to update the comment as well?

Comment (srinathgit) [reply marklogic-data-hub/src/main/java/com/marklogic/hub/impl/DataHubImpl.java]: Updated the comment

**Comment by srinathgit** [comment]
Sorry for multiple review requests, all the tests seems to have passed finally

**Pull request merged**